ethereum-consensus/src/capella/block_processing.rs

```diff
@@ -37,10 +41,33 @@ pub fn process_bls_to_execution_change<
         BYTES_PER_LOGS_BLOOM,
         MAX_EXTRA_DATA_BYTES,
     >,
-    _signed_address_change: &mut SignedBlsToExecutionChange,
-    _context: &Context,
+    signed_address_change: &mut SignedBlsToExecutionChange,
+    context: &Context,
 ) -> Result<()> {
-    unimplemented!()
+    let address_change = signed_address_change;
+
+    assert!(address_change.message.validator_index < state.validators.len());
+
+    let validator = &mut state.validators[address_change.message.validator_index];
+
+    assert!(validator.withdrawal_credentials.starts_with(&[BLS_WITHDRAWAL_PREFIX]));
+    assert!(
+        validator.withdrawal_credentials[1..]
+            == hash(address_change.message.from_bls_public_key.as_ref())[1..]
+    );
+
+    let domain =
+        compute_domain(BlsToExecutionChange, None, Some(state.genesis_validators_root), context)?;
+    let signing_root = compute_signing_root(address_change, domain)?;
+    let pk: &BlsPublicKey = &address_change.message.from_bls_public_key;
+    assert!(verify_signature(&pk, signing_root.as_ref(), &address_change.signature,).is_ok());
+    let withdrawal_credentials = vec![ETH1_ADDRESS_WITHDRAWAL_PREFIX];
+
+    let withdrawal_credentials_array: [u8; 32] =
+        withdrawal_credentials.try_into().expect("Wrong size");
+    validator.withdrawal_credentials = ByteVector::try_from(withdrawal_credentials_array.as_ref())
+        .expect("Failed to convert array to ByteVector");
```

Comment:
we are missing the actual address to set, and also we can work directly w/ the memory already allocated for the `withdrawal_credentials`:

```suggestion
    validator.withdrawal_credentials[0] = ETH1_ADDRESS_WITHDRAWAL_PREFIX;
    validator.withdrawal_credentials[1..12].fill(0);
    validator.withdrawal_credentials[12..].copy_from_slice(address_change.message.to_execution_address.as_ref());
```

Reply:
I couldnt do it this way , as it yields the following error 

```shell
`ssz_rs::Vector<u8, 32>` as mutable
trait `IndexMut` is required to modify indexed content, but it is not implemented for `ssz_rs::Vector<u8, 32>`
```

ssr_rs is an external library, so we can't apply `IndexMut` to it . 

I ended up having to cast it to a vec, abd then a `ByteVector<32>`.

```rust
    validator.withdrawal_credentials[0] = ETH1_ADDRESS_WITHDRAWAL_PREFIX;
    let mut withdrawal_credentials: Vec<u8> = validator.withdrawal_credentials.as_ref().to_vec();
    withdrawal_credentials[12..].copy_from_slice(address_change.to_execution_address.as_ref());
    let withdrawal_credentials: &[u8] = &withdrawal_credentials;
    let withdrawal_credentials =
        ByteVector::<32>::try_from(withdrawal_credentials).expect("Wrong size");
    validator.withdrawal_credentials = withdrawal_credentials;
  ```

I also had a question with regards to whether or not this was redudnant since we are creating an vector now

```rust
    validator.withdrawal_credentials[1..12].fill(0);
```

I have omitted it , but happy to add it back

Reply:
ah sorry about this, I ran into elsewhere and went ahead and patched it in `ssz_rs` here: https://github.com/ralexstokes/ssz-rs/pull/105

if you rebase from `main` you should have the updated fuctionality and should be able to do what I suggested above